src/main/java/focusedCrawler/link/LinkStorage.java

```diff
@@ -155,12 +159,41 @@ public Object insert(Page page) throws StorageException {
      */
     public synchronized Object select(Object obj) throws StorageException, DataNotFoundException {
         try {
-            return frontierManager.nextURL();
+            LinkRelevance link = frontierManager.nextURL();
+            if(!blackList.contains(link.getTopLevelDomainName())){
+                return link;
+            }else {
+                logger.info("Dead Domain ignored: "+link.getTopLevelDomainName());
+                return select(null);
+            }
         } catch (FrontierPersistentException e) {
             throw new StorageException(e.getMessage(), e);
         }
     }
 
+    public synchronized void addToBlackList(String url){
```

Comment:
I think only one of these 3 `addToBlackList` methods are used, you can remove the unused ones if they are not needed.